source/client-side-encryption/client-side-encryption.rst

```diff
@@ -708,20 +708,44 @@ ClientEncryption
 
       // Creates a new key document and inserts into the key vault collection.
       // Returns the \_id of the created document as a UUID (BSON binary subtype 4).
-      createKey(kmsProvider: String, opts: Optional<DataKeyOpts>): Binary;
+      createKey(kmsProvider: String, opts: Optional<DataKeyOpts>): UUID;
 
       // An alias function equivalent to createKey.
-      createDataKey(kmsProvider: String, opts: Optional<DataKeyOpts>): Binary;
+      createDataKey(kmsProvider: String, opts: Optional<DataKeyOpts>): UUID;
 
       // Decrypts multiple data keys and (re-)encrypts them with a new masterKey, or with their current masterKey if a new one is not given.
       // Returns a RewrapManyDataKeyResult.
       rewrapManyDataKey(filter: Document, opts: Optional<RewrapManyDataKeyOpts>): RewrapManyDataKeyResult;
 
+      // Removes the key document with the given \_id from the key vault collection.
+      // Returns the result of the internal deleteOne() operation on the key vault collection.
+      deleteKey(id: UUID): DeleteResult;
+
+      // Finds a single key document with the given UUID.
+      // Returns the result of the internal find() operation on the key vault collection.
+      getKey(id: UUID): Iterable<Document>;
+
+      // Finds all documents in the key vault collection.
+      // Returns the result of the internal find() operation on the key vault collection.
+      getKeys(): Iterable<Document>;
+
+      // Adds a keyAltName to the keyAltNames array of the key document in the key vault collection with the given \_id.
+      // Returns the previous version of the key document.
+      addKeyAlternateName(id: UUID, keyAltName: String): Optional<Document>;
+
+      // Removes a keyAltName from the keyAltNames array of the key document in the key vault collection with the given \_id.
+      // Returns the previous version of the key document.
+      removeKeyAlternateName(id: UUID, keyAltName: String): Optional<Document>;
+
+      // Returns a key document in the key vault collection with the given \_id.
+      getKeyByAltName(keyAltName: String): Iterable<Document>;
+
       // Encrypts a BSONValue with a given key and algorithm.
       // Returns an encrypted value (BSON binary of subtype 6). The underlying implementation may return an error for prohibited BSON values.
       encrypt(value: BSONValue, opts: EncryptOpts): Binary;
 
-      // Decrypts an encrypted value (BSON binary of subtype 6). Returns the original BSON value.
+      // Decrypts an encrypted value (BSON binary of subtype 6).
+      // Returns the original BSON value.
       decrypt(value: Binary): BSONValue;
```

Comment:
Is this the first use of "BSONValue" in a spec? Previous specs have used `any` (assuming that all language values have a BSON representation). I can imagine that the libmongoc API will use `bson_value_t`, but I'm curious about whether this will be clear for other languages.

Reply:
The `BsonValue` term in those specs seems to be the same thing that would apply here, so I'd suggest using that form of capitalization for consistency.